src/content/learn/typescript.md

```diff
@@ -0,0 +1,433 @@
+---
+title: Using TypeScript
+re: https://github.com/reactjs/react.dev/issues/5960
+---
+
+<Intro>
+
+TypeScript is a popular way to add type definitions to JavaScript codebases. Out of the box, TypeScript [supports JSX](/learn/writing-markup-with-jsx) and you can get full React support by adding [`@types/react-dom`](https://www.npmjs.com/package/@types/react-dom) to your project.
```

Comment:
`@types/react` is the main one since that's what you mostly interact with (hooks, utility types etc). `@types/react-dom` is onyl for `react-dom`